### PR TITLE
stop build if config file not exist

### PR DESCRIPTION
**Pull request by danielqsj**
Adding a feature: stop build if config file in `--config` is set but not exist

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/4308

### CODE REVIEW

**Comment by jpkrohling** [comment]
I'll review and do a manual test possibly later today, but wanted to ask whether it would be possible to add a test for this.

**Comment by jpkrohling** [comment]
Could you please solve the conflicts?

**Comment by danielqsj** [comment]
@jpkrohling Code rebased :)

**Review by jpkrohling — APPROVED** [review]

File: cmd/builder/cmd/root.go
Comment (bogdandrutu) [inline cmd/builder/cmd/root.go:121]:
```suggestion
	if err := vp.ReadInConfig(); err != nil {
		cobra.CheckErr(err)
	}
	cfg.Logger.Info("Using config file", zap.String("path", vp.ConfigFileUsed()))
```

Comment (danielqsj) [reply cmd/builder/cmd/root.go]: Agree. Code updated!

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Looks good to me, @danielqsj! While we wait for a maintainer to merge this, what do you think about trying to add a unit test for this? Do you think it's doable?

**Comment by bogdandrutu** [comment]
@jpkrohling no luck with tests

**Pull request merged**
